flight.py

```diff
@@ -0,0 +1,225 @@
+
+    Parameters
+    ----------
+    args : argparse.Namespace
+        Arguments assembled by argparse.
+    controller : flight.drone.drone_controller.DroneController
+        The second parameter.
+
+    Returns
+    -------
+    list
+        Blueprints to be added to Flask app
+
+    """
+    pass
+
+
+def debug_add_task(controller, command, meta):
+    if "priority" in meta:
+        priority = get_priority(meta["priority"])
+    if "direction" in meta:
+        direction = get_direction(meta["direction"])
+
+    if command == "exit":
+        controller.add_exit_task(c.Priorities.HIGH)
+        return False
+    elif command == "land":
+        controller.add_land_task(priority)
+    elif command == "hover":
+        controller.add_hover_task(fc.DEFAULT_ALTITUDE, int(meta["time"]),
+                                  priority)
+    elif command == "takeoff":
+        controller.add_takeoff_task(int(meta["altitude"]))
+    elif command == "move":
+        controller.add_linear_movement_task(direction, int(meta["time"]),
+                                            priority)
+    return True
+
+
+def get_priority(priority):
+    return get_enum(c.Priorities, priority)
+
+
+def get_direction(direction):
+    return get_enum(c.Directions, direction)
+
+
+def get_enum(enum, key):
+    item = key.upper()
+    if hasattr(enum, item):
+        return getattr(enum, item)
+    else:
+        raise InvalidDirectionException(
+            'Expected value for type {}, got {}.'.format(type(enum), item))
+
+
+def start_ai(controller, routine):
+    importlib.import_module("flight.AIs.{}".format(routine))
+
+
+def kill_ai(controller):
+    pass
+
+
+def list_tasks(controller):
+    return controller.queue
+
+
+def add_task(args, controller, data):
+    if args.debug:
+        # If in debug mode
+        if not ("command" in data and "meta" in data):
+            logging.error("command and meta required")
+            return True
+        command = data["command"].lower()
+        meta = data["meta"]
+        try:
+            return debug_add_task(controller, command, meta)
+        except (InvalidDirectionException, InvalidPriorityException) as err:
+            return err
+        except Exception as err:
+            logging.error("Unexpected error, killing drone" + str(err))
+            debug_add_task(controller, "exit", {})
+            return False
+    elif args.routine:
+        # If in production mode, starting or stopping drone is only option
+        # get push data and check if start or kill
+        command = data["command"].lower()
+        if command == "start":
+            # start routine
+            start_ai(controller, args.routine)
+        elif command == "kill":
+            # force into land
+            kill_ai(controller)
+
+        return "routine " + args.routine + " in progress"
+
+
+def tcp_thread(args, controller):
+    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    # set_keepalive_osx(s)
```

Comment:
Are the keepalive functions for the client? If so, would it be hard to add windows as an option? The laptop I always bring during testing sessions runs windows. Also, could the OS be a command line arg?

Reply:
For the drone, so there won't be any OS specific stuff.